src/cc/bcc_proc.c

```diff
@@ -82,6 +84,48 @@ int bcc_mapping_is_file_backed(const char *mapname) {
     STARTS_WITH(mapname, "[vsyscall]"));
 }
 
+/*
+Finds a file descriptor for a given inode if it's a memory-backed fd.
+*/
+static char *_procutils_memfd_path(int pid, uint64_t inum) {
+  char dir_buffer[1024];
+  char *path = NULL;
+  char *dirstr;
+  DIR *dirstream;
+  int dir_size;
+  struct stat sb;
+  struct dirent *dent;
+
+  snprintf(dir_buffer, 1024, "/proc/%d/fd/", pid);
+  dir_size = strlen(dir_buffer);
+  dirstr = calloc(sizeof(char), (dir_size + 1));
+  strcpy(dirstr, dir_buffer);
+  dirstream = opendir(dirstr);
+
+  while (path == NULL && dirstream != NULL &&
+         (dent = readdir(dirstream)) != NULL) {
+    char *pid_fd_path;
+    pid_fd_path =
+        calloc(sizeof(char), strlen(dirstr) + strlen(dent->d_name) + 1);
+    strcat(pid_fd_path, dirstr);
+    strcat(pid_fd_path, dent->d_name);
+
+    if (stat(pid_fd_path, &sb) == -1) {
+      free(pid_fd_path);
+      continue;
```

Comment:
I decided there is no point in printing an error/warning if we can't stat an inode, we'll just return NULL and let the caller handle it i we don't end up finding what we need. No need to print errors for fds that don't matter.